webhook/resources/backingimage/mutator.go

```diff
@@ -91,6 +93,27 @@ func (b *backingImageMutator) Create(request *admission.Request, newObj runtime.
 		}
 	}
 
+	if longhorn.BackingImageDataSourceType(backingImage.Spec.SourceType) == longhorn.BackingImageDataSourceTypeClone {
+		// inherit the secret and secretNamespace when cloning from another backing image and the encryption is ignore
+		if bimtypes.EncryptionType(parameters[longhorn.DataSourceTypeCloneParameterEncryption]) == bimtypes.EncryptionTypeIgnore {
+			sourceBackingImageName := parameters[longhorn.DataSourceTypeCloneParameterBackingImage]
+			sourceBackingImage, err := b.ds.GetBackingImageRO(sourceBackingImageName)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to get source backing image %v", sourceBackingImageName)
+			}
+			patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/secret", "value": "%s"}`, sourceBackingImage.Spec.Secret))
+			patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/secretNamespace", "value": "%s"}`, sourceBackingImage.Spec.SecretNamespace))
+			if sourceBackingImage.Status.Checksum == "" {
+				return nil, errors.Wrapf(err, "failed to get checksum of source backing image %v", sourceBackingImageName)
+			}
```

Comment:
Do we need the check if checksum is empty?

Reply:
I think we can always override the checksum no matter if it is empty